docs/API.md

```diff
@@ -801,38 +808,38 @@ Example response:
 
 An object with the following properties:
 
-* `"username"`: A string, optional. The name of the user to add.
-* `"api_key"`: A string, optional. The API key (in effect, password) for the user to add.
-
->
-> TODO: Are `"username"` and `"api_key"` really supposed to be optional? Because
-> they are now, is that a bug? I get `"result": 0` when I omit them. I would
-> have expected parameter validation errors.
->
-
+* `"username"`: A string, required. The name of the user to add.
+* `"api_key"`: A string, required. The API key (in effect, password) for the user to add.
```

Comment:
We should put a limit of what character to use here. And lenght.

Reply:
`api_key` should specify a data type and the data type should specify a regular expression.